File: Lab9ZastavniyKI305/KI305/Zastavnyi/lab9/Leaf.py
class Leaf:
    def __init__(self, shape="Немає", state=False, color="Немає"):
        """
        Конструктор класу Leaf.

        Ініціалізує атрибути листка рослини.

        Args:
        - shape (str): Форма листка.
        - state (bool): Стан листка (присутній/відсутній).
        - color (str): Колір листка.

        """
        self.shape = shape
        self.state = state
        self.color = color

    def get_shape(self):
        """
        Отримання форми листка.

        Returns:
        - str: Форма листка.

        """
        return self.shape

    def set_shape(self, shape):
        """
        Встановлення форми листка.

        Args:
        - shape (str): Форма листка.

        """
        self.shape = shape

    def get_state(self):
        """
        Отримання стану листка.

        Returns:
        - bool: Стан листка (присутній/відсутній).

        """
        return self.state

    def set_state(self, state):
        """
        Встановлення стану листка.

        Args:
        - state (bool): Стан листка (присутній/відсутній).

        """
        self.state = state

    def get_color(self):
        """
        Отримання коліру листка.

        Returns:
        - str: Колір листка.

        """
        return self.color

    def set_color(self, color):
        """
        Встановлення коліру листка.

        Args:
        - color (str): Колір листка.

        """
        self.color = color

    def get_status(self, fout):
        """
        Виведення статусу листка.

        Args:
        - fout (File): Файловий об'єкт для запису.

        """
        if self.state:
            print("Листя: Є")
            fout.write("Листя: Є\n")
        else:
            print("Листя: Немає")
            fout.write("Листя: Немає\n")

        print("Форма:", self.shape)
        print("Колір:", self.color)

        fout.write("Форма: " + self.shape + "\n")
        fout.write("Колір: " + self.color + "\n")

File: Lab2ZastavnyiKI305/Lab2ZastavnyiKI305/src/KI305/Zastavnyi/Lab2/Leaf.java
package KI305.Zastavnyi.Lab2;

import java.io.PrintWriter;

import static java.lang.System.out;

/**
 * Клас, що представляє об'єкт "Лист". Кожен лист може мати форму, стан (наявний або відсутній) та колір.
 * Для спрощення можливості створення об'єктів за замовчуванням та з заданими параметрами є конструктори.
 */
public class Leaf {
    private String shape;  // Форма листя
    private boolean state; // Наявність листя (наявний - true, відсутній - false)
    private String color;  // Колір листя

    /**
     * Конструктор за замовчуванням. Встановлює всі властивості листя на значення "Немає".
     */
    public Leaf() {
        this.shape = "Немає";
        this.state = false;
        this.color = "Немає";
    }

    /**
     * Конструктор, який дозволяє встановити стан листя при створенні об'єкта.
     *
     * @param state Стан листя (наявний - true, відсутній - false)
     */
    public Leaf(boolean state) {
        this.shape = "Немає";
        this.state = state;
        this.color = "Немає";
    }

    /**
     * Конструктор, який дозволяє встановити форму, стан та колір листя при створенні об'єкта.
     *
     * @param shape Форма листя
     * @param state Стан листя (наявний - true, відсутній - false)
     * @param color Колір листя
     */
    public Leaf(String shape, boolean state, String color) {
        this.shape = shape;
        this.state = state;
        this.color = color;
    }

    /**
     * Метод, що повертає форму листя.
     *
     * @return Форма листя
     */
    public String getShape() {
        return shape;
    }

    /**
     * Метод, що встановлює форму листя.
     *
     * @param shape Форма листя
     */
    public void setShape(String shape) {
        this.shape = shape;
    }

    /**
     * Метод, що повертає стан листя.
     *
     * @return Стан листя (наявний - true, відсутній - false)
     */
    public boolean getState() {
        return state;
    }

    /**
     * Метод, що встановлює стан листя.
     *
     * @param state Стан листя (наявний - true, відсутній - false)
     */
    public void setState(boolean state) {
        this.state = state;
    }

    /**
     * Метод, що повертає колір листя.
     *
     * @return Колір листя
     */
    public String getColor() {
        return color;
    }

    /**
     * Метод, що встановлює колір листя.
     *
     * @param color Колір листя
     */
    public void setColor(String color) {
        this.color = color;
    }

    /**
     * Метод, що виводить статус листя на консоль та у файл за допомогою PrintWriter.
     *
     * @param fout PrintWriter об'єкт для запису у файл
     */
    public void getStatus(PrintWriter fout) {
        if (state) {
            out.println("Листя: Є");
            fout.println("Листя: Є");
        } else {
            out.println("Листя: Немає");
            fout.println("Листя: Немає");
        }
        out.println("Форма: " + this.shape);
        out.println("Колір: " + this.color);

        fout.println("Форма: " + this.shape);
        fout.println("Колір: " + this.color);
    }
}
